### PR TITLE
Enable getcoursechapter with its dependent schemas

### PR DESCRIPTION
**Pull request by yilmaz-burak**
I have implemented create and read functionalities for chapter,content,discussion,comment,badge schemas. Would be nice if you could have a look, @furkanakkurt1335 @ersoykadir @Goko6 . Thanks!

### CODE REVIEW

File: app/server/src/routes/badge/badge.validate.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const { body, query } = require("express-validator");
+
+exports.validate = (method) => {
+    switch (method) {
+
+        case "getCall":
```
Comment (ersoykadir) [inline app/server/src/routes/badge/badge.validate.js:6]: We might want validation for post requests, at least for existence. In this case, createBadge

File: app/server/src/routes/badge/badge.controller.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const BadgeController = {
+    createBadge: async function (req, res) {
+        try {
+            const { title, description } = req.body
+            const badge = await BadgeModel.createBadge(
+                title,
+                description,
+            )
+            res.send({ "status": "ok", "message": badge })
```
Comment (ersoykadir) [inline app/server/src/routes/badge/badge.controller.js:11]: We maybe set create functions return-status codes as 201.

File: app/server/src/routes/chapter/chapter.controller.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+const ChapterController = {
+    createChapter: async function (req, res) {
+        try {
+            const { chapter_name, chapter_badge_id, content_id_list } = req.body
+            const chapter = await ChapterModel.createChapter(
+                chapter_name,
+                chapter_badge_id,
+                content_id_list,
+            )
+            res.send({ "status": "ok", "message": chapter })
```
Comment (ersoykadir) [inline app/server/src/routes/chapter/chapter.controller.js:12]: Similarly, can be status code 201

File: app/server/src/routes/chapter/chapter.validate.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const { body, query } = require("express-validator");
+
+exports.validate = (method) => {
+    switch (method) {
+
+        case "getCall":
```
Comment (ersoykadir) [inline app/server/src/routes/chapter/chapter.validate.js:6]: Similarly, createChapter can have req body existence validations

File: app/server/src/routes/comment/comment.controller.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    createComment: async function (req, res) {
+        try {
+            const { user_id, comment_body, comment_files } = req.body
+
+            const comment = await CommentModel.createComment(
+                user_id,
+                comment_body,
+                comment_files,
+            )
+            res.send({ "status": "ok", "message": comment })
```
Comment (ersoykadir) [inline app/server/src/routes/comment/comment.controller.js:14]: status code 201

File: app/server/src/routes/comment/comment.validate.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const { body, query } = require("express-validator");
+
+exports.validate = (method) => {
+    switch (method) {
+
+        case "getCall":
```
Comment (ersoykadir) [inline app/server/src/routes/comment/comment.validate.js:6]: createComment validation

File: app/server/src/routes/content/content.controller.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        try {
+            const { name, body, media, discussion_id, video_link } = req.body
+            const badge = await ContentModel.createContent(
+                name,
+                body,
+                media,
+                discussion_id,
+                video_link,
+            )
+            res.send({ "status": "ok", "message": badge })
```
Comment (ersoykadir) [inline app/server/src/routes/content/content.controller.js:14]: status code 201

File: app/server/src/routes/content/content.validate.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+const { body, query } = require("express-validator");
+
+exports.validate = (method) => {
+    switch (method) {
+
+        case "getCall":
```
Comment (ersoykadir) [inline app/server/src/routes/content/content.validate.js:6]: createContent validation

File: app/server/src/routes/discussion/discussion.controller.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            const { user_id, course_id, comment_id_list, discussion_body, discussion_files } = req.body
+
+            const discussion = await DiscussionModel.createDiscussion(
+                user_id,
+                course_id,
+                comment_id_list,
+                discussion_body,
+                discussion_files,
+            )
+            res.send({ "status": "ok", "message": discussion })
```
Comment (ersoykadir) [inline app/server/src/routes/discussion/discussion.controller.js:15]: status code 201

File: app/server/src/routes/discussion/discussion.validate.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+const { body, query } = require("express-validator");
+
+exports.validate = (method) => {
+    switch (method) {
+
+        case "getCall":
```
Comment (ersoykadir) [inline app/server/src/routes/discussion/discussion.validate.js:6]: createDiscussion validation

**Review by ersoykadir — COMMENTED** [review]
Well done. Good to go after small updates.

File: app/server/src/models/contents/contents.model.js [hunk 11]
```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose")
+
+const contentSchema = new mongoose.Schema({
+    name: String,
+    body: String,
+    media: [String],
+    discussion: { type: mongoose.Schema.Types.ObjectId, ref: "Discussion" },
+    video: String,
```
[the remark this answers is not in the document]
Comment (yilmaz-burak) [reply app/server/src/models/contents/contents.model.js:8]: We will update the class diagrams... Specifications required this.

File: app/server/src/models/chapters/chapters.model.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    .populate("chapter_badge content chapter_name")
+    .populate({
+      path: 'content',
+      populate: {
+        path: 'discussion',
+        populate: { path: 'comments' }
+      }
+    })
+    .exec()
+}
```
Comment (furkanakkurt1335) [inline app/server/src/models/chapters/chapters.model.js:35]: why is there a need for populating content downwards here?

Comment (yilmaz-burak) [reply app/server/src/models/chapters/chapters.model.js:35]: we also need to see discussion and its inner schema infos

**Pull request merged**